### PR TITLE
feat: support DROP MEASUREMENT in specific retention policy

### LINKED ISSUE #639: `drop measurement` feature is incomplete
### Describe the bug（Bug 描述）

when two or more measurements with same name are created in different retention policies within a same database, the `drop measurement` statement always drop the measurement in the default retention policy first.

### To Reproduce（Bug 复现步骤）

```shell
> create database db0
> use db0
> create retention policy rp3d on db0 duration 3d replication 1
> insert cpu,host=server01 usage=10
> use db0.rp3d
> insert cpu,host=server02 usage=20

> select * from rp3d.cpu
name: cpu
+---------------------+----------+-------+
|        time         |   host   | usage |
+---------------------+----------+-------+
| 1719458520966886965 | server02 |    20 |
+---------------------+----------+-------+
3 columns, 1 rows in set

> select * from autogen.cpu
name: cpu
+---------------------+----------+-------+
|        time         |   host   | usage |
+---------------------+----------+-------+
| 1719458482991266362 | server01 |    10 |
+---------------------+----------+-------+
3 columns, 1 rows in set

> drop measurement cpu
> select * from rp3d.cpu
name: cpu
+---------------------+----------+-------+
|        time         |   host   | usage |
+---------------------+----------+-------+
| 1719458520966886965 | server02 |    20 |
+---------------------+----------+-------+
3 columns, 1 rows in set

> select * from autogen.cpu
ERR: measurement not found
> drop measurement cpu
> select * from rp3d.cpu
ERR: measurement not found
```

### Expected behavior（期望结果）

I can drop measurement in non-default retention polices and keep the measurement of the default retention policy with the same name unchanged.

### Screenshots（屏幕截图）

_No response_

### Logs（完整的错误日志）

_No response_

### Additional context（其他的一些补充内容）

_No response_

### PR DESCRIPTION
**Pull request by StepY1aoZz**
### What problem does this PR solve?

resolve #639 

### What is changed and how it works?
Now in the `drop measurement` command users can specify a retention policy when deleting measurements. If the retention policy is not specified in command, the default retention policy of current database will be used.
This feat is useful when users have many measurements in different retention policies but with same  measurement name.

Example:
```
// drop measurement in specified retention policy
DROP MEASUREMENT <retention_policy_name>.<measurement_name>

// drop measurement in default retention policy
DROP MEASUREMENT <measurement_name>
``` 

### How Has This Been Tested?
A new test case was added into server_test.go.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: lib/metaclient/meta_client.go [hunk 1]
```diff
@@ -187,7 +187,7 @@ type MetaClient interface {
 	Users() []meta2.UserInfo
 	MarkDatabaseDelete(name string) error
 	MarkRetentionPolicyDelete(database, name string) error
-	MarkMeasurementDelete(database, mst string) error
+	MarkMeasurementDelete(database, mst, rpName string) error
```
Comment (localvar) [inline lib/metaclient/meta_client.go:190]: propose change the order of the parameters to make it align with other functions(like `Measurement` below).

```suggestion
	MarkMeasurementDelete(database, rpName, mst string) error
```

Comment (localvar) [inline lib/metaclient/meta_client.go:190]: propose change the order of the parameters to make it align with other functions(like `Measurement` below).

```suggestion
	MarkMeasurementDelete(database, rpName, mst string) error
```

Comment (localvar) [inline lib/metaclient/meta_client.go:190]: propose change the order of the parameters to make it align with other functions(like `Measurement` below).

```suggestion
	MarkMeasurementDelete(database, rpName, mst string) error
```

File: lib/metaclient/meta_client.go [hunk 2]
```diff
@@ -1278,16 +1278,17 @@ func (c *Client) MarkMeasurementDelete(database, measurement string) error {
 	}
 
 	var policy string
-	for _, rp := range dbi.RetentionPolicies {
+	if rpName == "" {
```
Comment (localvar) [inline lib/metaclient/meta_client.go:1281]: `rpName` or `policy` could be removed to simpilify the code.

File: app/ts-store/run/server_test.go [hunk 3]
```diff
@@ -385,7 +385,7 @@ func (client *MockMetaClient) MarkDatabaseDelete(name string) error {
 func (client *MockMetaClient) MarkRetentionPolicyDelete(database, name string) error {
 	return nil
 }
-func (client *MockMetaClient) MarkMeasurementDelete(database, mst string) error {
+func (client *MockMetaClient) MarkMeasurementDelete(database, mst, rpName string) error {
```
Comment (xiangyu5632) [inline app/ts-store/run/server_test.go:388]: instead 'rpName' with rpname may be better, or 'rp_name'

**Review by xiangyu5632 — APPROVED** [review]

File: app/ts-store/run/server_test.go (hunk 3 above)
Comment (localvar) [reply app/ts-store/run/server_test.go:388]: `rp_name` should not be used, refer https://google.github.io/styleguide/go/decisions#underscores.
`rpName` and `rpname` is acceptable, but for consistency with `database` and `mst`, `rp` or `retentionPolicy` is better, and `policy` is already used in the code, it is also an option.

**Comment by StepY1aoZz** [comment]
All the feedbacks are incorporated in the latest commit, including:

1. changed the order of arguments to `database,policy,measurement`
2. rename the argument `rpName` to `policy`
3. simplified the logic in func `MarkMeasurementDelete`

Thanks for these useful suggestions. @localvar @xiangyu5632

**Review by xiangyu5632 — APPROVED** [review]

**Pull request merged**